### PR TITLE
Show sponsorship placeholder when adblock active

### PR DESCRIPTION
**Pull request by tijanamiletic**
Moving the sponsorship placeholder outside the DFP container when the adblock is detected to make sure the sponsorship information is not hidden by the adblock.

Applies in cases when the sponsorship badge is not inside a larger DFP content container.

Result

Without adblock:

screen shot 2015-08-20 at 15 18 35

With adblock:

screen shot 2015-08-20 at 15 18 16

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 1]
```diff
@@ -157,6 +157,15 @@ define([
             return config.switches.mobileTopBannerRemove && $('.top-banner-ad-container--ab-mobile').length > 0 && detect.getBreakpoint() === 'mobile';
         },
 
+        isSponsorshipContainerTest = function () {
+            var sponsorshipIds = ['#dfp-ad--adbadge', '#dfp-ad--spbadge', '#dfp-ad--fobadge', '#dfp-ad--adbadge1', '#dfp-ad--spbadge1', '#dfp-ad--fobadge1'];
+            for (var i = 0; i < sponsorshipIds.length; i++) {
+                if ($(sponsorshipIds[i]).length > 0) {
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: you can just check "if ($(sponsorshipIds[i]).length)"

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 2]
```diff
@@ -157,6 +157,15 @@ define([
             return config.switches.mobileTopBannerRemove && $('.top-banner-ad-container--ab-mobile').length > 0 && detect.getBreakpoint() === 'mobile';
         },
 
+        isSponsorshipContainerTest = function () {
+            var sponsorshipIds = ['#dfp-ad--adbadge', '#dfp-ad--spbadge', '#dfp-ad--fobadge', '#dfp-ad--adbadge1', '#dfp-ad--spbadge1', '#dfp-ad--fobadge1'];
+            for (var i = 0; i < sponsorshipIds.length; i++) {
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: use lodash forEach https://lodash.com/docs#forEach

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 3]
```diff
@@ -254,6 +263,14 @@ define([
             // anything we want to happen after displaying ads
             window.googletag.cmd.push(postDisplay);
 
+            // show sponsorship placeholder if adblock detected
+            var sponsorshipIdFound = isSponsorshipContainerTest();
+            var sponsorshipIdClasses = $(sponsorshipIdFound).attr('class').replace('ad-slot ', '');
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: use one var statement

Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: you have to search the DOM for $(sponsorshipIdFound) few times here (any operation in DOM is slow). Maybe assign it to the variable? Also remember to use fastdom

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 4]
```diff
@@ -254,6 +263,14 @@ define([
             // anything we want to happen after displaying ads
             window.googletag.cmd.push(postDisplay);
 
+            // show sponsorship placeholder if adblock detected
+            var sponsorshipIdFound = isSponsorshipContainerTest();
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: I would move this whole block to another function

Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: Do you just want to do this on the first element with a sponsorship ID, or would we just want to perform this operation against every match on the page? Just a thought.

**Comment by tijanamiletic** [comment]
Suggested changes have been implemented. This code only works when there is a placeholder with sponsorship information in place (which gets overwritten by the DFP call if there is no active adblock). No cases have been found where there is more than one sponsorship badge placeholder per page.

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 5]
```diff
@@ -157,6 +157,40 @@ define([
...
 
+        isSponsorshipContainerTest = function () {
+            var sponsorshipIds = ['#dfp-ad--adbadge', '#dfp-ad--spbadge', '#dfp-ad--fobadge', '#dfp-ad--adbadge1', '#dfp-ad--spbadge1', '#dfp-ad--fobadge1', '#dfp-ad--adbadge2', '#dfp-ad--spbadge2', '#dfp-ad--fobadge2', '#dfp-ad--adbadge3', '#dfp-ad--spbadge3', '#dfp-ad--fobadge3', '#dfp-ad--adbadge4', '#dfp-ad--spbadge4', '#dfp-ad--fobadge4', '#dfp-ad--adbadge5', '#dfp-ad--spbadge5', '#dfp-ad--fobadge5'],
+                sponsorshipIdsReturned = [],
+                sponsorshipIdReturned;
+
+            _.forEach(sponsorshipIds, function (value) {
+                if ($(value).length) {
+                    sponsorshipIdReturned = value;
+                    sponsorshipIdsReturned.push(sponsorshipIdReturned);
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: Do you need `sponsorshipIdReturned`? You can do just `sponsorshipIdsReturned.push(value);`

**Comment by tijanamiletic** [comment]
Changes have been made

**Comment by Calanthe** [comment]
:+1:

**Pull request merged**